packages/angular_devkit/build_angular/src/tools/sass/rebasing-importer.ts

```diff
@@ -22,6 +23,38 @@ export interface DirectoryEntry {
   directories: Set<string>;
 }
 
+/**
+ * A prefix that is added to import and use directive specifiers that should be resolved
+ * as modules and that will contain added resolve directory information.
+ *
+ * This functionality is used to workaround the Sass limitation that it does not provide the
+ * importer file to custom resolution plugins.
+ */
+const MODULE_RESOLUTION_PREFIX = '__NG_PACKAGE__';
+
+function packModuleSpecifier(specifier: string, resolveDir: string): string {
+  const packed = MODULE_RESOLUTION_PREFIX + ';' + resolveDir + ';' + specifier;
+
+  // Normalize path separators and escape characters
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/url#syntax
+  const normalizedPacked = packed.replace(/\\/g, '/').replace(/[()\s'"]/g, '\\$&');
+
+  return normalizedPacked;
+}
+
+function unpackModuleSpecifier(specifier: string): { specifier: string; resolveDir?: string } {
+  if (specifier.startsWith(`${MODULE_RESOLUTION_PREFIX};`)) {
```

Comment:
Nit: The "interesting" part of the function is the `specifier.split(';', 3)`. I suggest inverting the `if` statement to early return `{ specifier }` in the uninteresting case so the rest of the function can focus on the interesting one.